js/renderer.js

```diff
@@ -115,8 +115,8 @@ class Renderer {
         this.initBullets(data[2]);
         this.initShips(data[3]);
 
-        this.lastTime = -1
-        requestAnimationFrame((ts) => this.renderLoop(ts));
+        this.lastTime = performance.now();
```

Comment:
This does not work. If you console.log(this.lastTime) vs console.log(ts) in the first iteration of renderLoop, you will see that this.lastTime is over 100ms later than the given ts. They must be getting their timestamp from different sources or at different times.

Reply:
First of all, with the way you've implemented your moving average, the first 10 frames are going to be a progressively better estimate, but the first 2 or 3 are still quite wrong. That's fine as far as I'm concerned, but could probably be better with something that takes into account the number of elements actually added to the list.

But now that I play with this a little more, I understand better what you are saying. Basically there is a [bug in how this works](https://stackoverflow.com/questions/38360250/requestanimationframe-now-vs-performance-now-time-discrepancy). Also worth noting, Firefox broke the spec for a lot of this to mitigate spectre, while other browsers did not. For example `performance.now()` in FF is always truncated, while in Chrome it's got decimal points.

Fuck I hate web development.